src/snowflake/connector/auth.py

```diff
@@ -109,13 +111,50 @@
 
 ID_TOKEN = "ID_TOKEN"
 MFA_TOKEN = "MFATOKEN"
+DEFAULT_MAX_CNXN_RETRY_ATTEMPTS = 1
+
+
+class AuthRetryCtx:
+    def __init__(self):
+        self._current_retry_count = 0
+        self._max_retry_attempts = int(
+            getenv("MAX_CNXN_RETRY_ATTEMPTS", DEFAULT_MAX_CNXN_RETRY_ATTEMPTS)
+        )
+        self._backoff = DecorrelateJitterBackoff(1, 16)
+        self._current_sleep_time = 1
+
+    def get_current_retry_count(self) -> int:
+        return self._current_retry_count
+
+    def increment_retry(self) -> None:
+        self._current_retry_count += 1
```

Comment:
I understand but I want to restrict what can be done on and to  `_current_retry_count`

Reply:
But since you are prepending a `_` you're signaling that other classes should not be interacting with it.
Like other classes can still just modify `self._current_retry_count` even if we add these getter and setter.

Reply:
Sure but having these specific functions (to be precise there's not setter there's just an increment function) communicates the intended actions on the data.